src/controller.cpp

```diff
@@ -529,14 +522,13 @@ Controller::UpdateResult Controller::update(
   ROS_DEBUG("d_end_phase: %f", d_end_phase);
   ROS_DEBUG("distance_to_goal: %f", distance_to_goal_);
 
-  // Get 'angle' towards current_goal
-  tf2::Transform base_to_goal = current_tf.inverseTimes(current_goal_);
-  const double angle_to_goal = atan2(base_to_goal.getOrigin().x(), -base_to_goal.getOrigin().y());
+  const auto in_direction_of_goal =
+    !std::signbit(current_tf.getOrigin().dot(current_goal_.getOrigin()));
 
```

Comment:
I like the seeming simplicity of this, but `current_tf` and `current_goal_` are both defined in the `map` frame, so you're now projecting the vector from `map` to `current_tf` onto the vector from `map` to `current_goal_`.  This doesn't do the same as in the original calculation.